routing/localchans/manager.go

```diff
@@ -309,7 +309,21 @@ func (r *Manager) createEdge(channel *channeldb.OpenChannel,
 // updateEdge updates the given edge with the new schema.
 func (r *Manager) updateEdge(tx kvdb.RTx, chanPoint wire.OutPoint,
 	edge *models.ChannelEdgePolicy,
-	newSchema routing.ChannelPolicy) error {
+	newSchema routing.ChannelPolicy) (*models.ChannelEdgePolicy, error) {
```

Comment:
No need to return the policy here.

Reply:
I disagree

- the processChan func can get a nil `edge`
- inside `r.updateEdge` the edge is created if it's nil, but not outside the updateEdge function
- Therefore, [line 121](https://github.com/lightningnetwork/lnd/pull/8805/files#diff-888432a99a4328c46ba85b63e86398ee6a79fc4fc4da59f7a30a1db88b2ddf91R121) would panic if `edge` would not be returned by tge `updateEdge` function.

Reply:
ahh correct, so we need to init a pointer to the edge before creating it then ? Feels kind of strange to pass the pointer and return the pointer ?

Reply:
or double pointer ?

Reply:
We could update the pointer value inside `updateEdge` too, like so:

```go
	if edge == nil {
		_, e, err := r.createEdge(channel, time.Now())
		if err != nil {
			return nil, err
		}

		*edge = *e
	}
```

Reply:
But then it may be less obvious that you cannot use `edge` _before_ calling `updateEdge`

Reply:
https://go.dev/play/p/5zRZFncWWvP

I think we cannot do your second proposal, the only way would prob. be double pointers.

Reply:
I tried this, which does work:
https://go.dev/play/p/-zHiy4RmsEN

Reply:
hmm but in our case edge is nil ?

so you example should begin with:

var thing *Thing => which makes thing `nil`

Reply:
Good point! So let's keep it like this?

Reply:
I would say let's keep it as is, double pointer probably are not super readable.

Reply:
Hi @JssDWt talked to the team, and we really don't wanna go with this approach of have a pointer as a function parameter and then also return a pointer. 

Can we not do something like this:

```
func (r *Manager) updateEdge(tx kvdb.RTx, chanPoint wire.OutPoint,
	edge models.ChannelEdgePolicy,
	newSchema routing.ChannelPolicy) (*models.ChannelEdgePolicy, error) {
```

so now we can in `processChan` do

if edge == nil {
edgee = _, edge, err = r.createEdge(channel, time.Now())
		if err != nil {
			return nil, err
		}
}

and then just do: `edge, err := r.updateEdge(tx, info.ChannelPoint, edge, newSchema)`


cc @ProofOfKeags can you confirm ?


}

Reply:
@JssDWt can we change the design here.